allennlp/commands/train.py

```diff
@@ -418,23 +452,24 @@ def _train_worker(
         batch_weight_key=batch_weight_key,
     )
 
-    try:
-        if distributed:  # let the setup get ready for all the workers
-            dist.barrier()
-
-        metrics = train_loop.run()
-    except KeyboardInterrupt:
-        # if we have completed an epoch, try to create a model archive.
-        if master and os.path.exists(os.path.join(serialization_dir, _DEFAULT_WEIGHTS)):
-            logging.info(
-                "Training interrupted by the user. Attempting to create "
-                "a model archive using the current best epoch weights."
-            )
-            archive_model(serialization_dir)
-        raise
-
-    if master:
-        train_loop.finish(metrics)
+    if not dry_run:
```

Comment:
No need to make this entire block nested - just do:

```
if dry_run:
	return None
```
above it.

Reply:
But if `dry_run` is `True`, I want the rest of the code to be executed (return the model if it's not distributed training):

https://github.com/allenai/allennlp/pull/3751/files/8c47ff48a1fbca1fde99940270e2ad0ff60227e4#diff-d4e8f335bf383d8d2b18287949003bbeR471-R472

Reply:
I'm not sure there is there a need to return the model? The functions above don't do anything with it, so I think we can return `None` and remove this nesting.

Reply:
It's transitively used here, for example:

https://github.com/allenai/allennlp/blob/57cdffb45205f1eeeff0737e9f3bb8f57103a29f/tutorials/tagger/config_allennlp.py#L91

Or you mean just if `dry_run == True`? If that's what you mean, I'm not sure of breaking the function contract in that case only, but we should state so in the docstring if we do.

Reply:
The thing is, we probably don't ever really want to return an untrained model - so this is actually saving people time (e.g if they forget to set dry-run to false again in some script, it's probably better that it returns None than an untrained model).